test/functional/Java8andUp/src/org/openj9/test/string/StringIndexOfString.java

```diff
@@ -0,0 +1,236 @@
+				subStringByteLen = (VECTOR_SIZE + 2);
+				break;
+			case 11:
+				// sub-string head aligned to byte 0 of source string. Length = VECTOR_SIZEx2
+				subStringByteIndex = 0;
+				subStringByteLen = VECTOR_SIZE * 2;
+				break;
+			case 12:
+				// sub-string head aligned to byte 0 of source string. Length = VECTOR_SIZEx2 +
+				// 4B
+				subStringByteIndex = 0;
+				subStringByteLen = VECTOR_SIZE * 2 + 4;
+				break;
+
+			case 13:
+				// 1st char collision
+				// sub-string is 3*VECTOR_SIZE long, spanning bettween the 4th and 7th 16B
+				// boundary.
+				subStringByteIndex = VECTOR_SIZE * 3;
+				subStringByteLen = VECTOR_SIZE * 3;
+				break;
+			case 14:
+				// s2 header collision.
+				// sub-string length > 3*VECTOR_SIZE long, starting from the 8th boundary and
+				// goes slightly pass the 11th boundary.
+				subStringByteIndex = (VECTOR_SIZE * 7);
+				subStringByteLen = (VECTOR_SIZE * 3 + 6);
+				break;
+			case 15:
+				// s2 header collision.
+				// sub-string length > 6*VECTOR_SIZE long. head and tail are not aligned to
+				// boundaries.
+				subStringByteIndex = (VECTOR_SIZE * 7 + 8);
+				subStringByteLen = (VECTOR_SIZE * 6 + 6);
+				break;
+			case 16:
+				// s2 header match but body mismatch
+				subStringByteIndex = 350;
+				subStringByteLen = 102;
+				break;
+			case 17:
+				// case 16. sub-string to be modified.
+				// test case for string not found. mismatch at the beginning
+				subStringByteIndex = 350;
+				subStringByteLen = 102;
+				break;
+			case 18:
+				// same as case 15. sub-string to be modified.
+				// test case for string not found. mismatch in the middle
+				subStringByteIndex = (VECTOR_SIZE * 7 + 8);
+				subStringByteLen = (VECTOR_SIZE * 6 + 6);
+				break;
+			case 19:
+				// same as case 15, sub-string to be modified.
+				// test case for string not found. mismatch at the end
+				subStringByteIndex = (VECTOR_SIZE * 7 + 8);
+				subStringByteLen = (VECTOR_SIZE * 6 + 6);
+				break;
+			default:
+				AssertJUnit.assertTrue("unexpected testcase number in test initialization", false);
+				return;
+			}
+
+			subStringByteIndex /= 2;
+			subStringByteLen /= 2;
+
+			String subString = searchString.substring(subStringByteIndex, subStringByteIndex + subStringByteLen);
+			int expectedResult = subStringByteIndex;
+
+			if (testCaseNum == 17) {
+				subString = "::" + subString;
+				expectedResult = -1;
+			} else if (testCaseNum == 18) {
+				int splicePoint = (int) (subString.length() * 0.6);
+				String sub1 = subString.substring(0, splicePoint - 1);
+				String sub2 = subString.substring(splicePoint + 2, subString.length() - 1);
+				subString = sub1 + "::" + sub2;
+				expectedResult = -1;
+			} else if (testCaseNum == 19) {
+				subString += "::";
+				expectedResult = -1;
+			}
+
+			expectedResults[testCaseNum - 1] = expectedResult;
+			inputStrings[testCaseNum - 1] = subString;
+		}
+	}
+
+	@Test
+	public void testStringIndexOfString() {
+		generateInputs();
```

Comment:
I don't see anything in `generateInputs` that requires runtime input, hence we should be moving it to a static constructor so these inputs are only generated once, not once per test since this test may be executed many, many times within the test bucket.

Reply:
Changed `generateInputs()`  to a static block.
This is a sanity test and all test functions annotated by `@Test` will be executed only once. Hence, this `generateInputs` function is only invoked once. So static block or not makes no difference here.